### PR TITLE
[Keyboard] add Allison keyboard

### PR DESCRIPTION
**Pull request by yiancar**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/allison/allison.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+/* This is a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT( \
```
Comment (noroadsleft) [inline keyboards/allison/allison.h]:
```suggestion
#define LAYOUT_all( \
```

File: keyboards/allison/info.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "allison",
+    "url": "www.yiancar-designs.com",
+    "maintainer": "Yiancar",
+    "width": 16.25, 
+    "height": 6.25, 
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/allison/info.json]:
```suggestion
        "LAYOUT_all": {
```

File: keyboards/allison/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT( /* Base */
```
Comment (noroadsleft) [inline keyboards/allison/keymaps/default/keymap.c]:
```suggestion
    [0] = LAYOUT_all( /* Base */
```

File: keyboards/allison/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+             KC_ESC,  KC_F1,   KC_F2,   KC_F3, KC_F4, KC_F5, KC_F6, KC_F7,  KC_F8, KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_INS,
+    KC_HOME, KC_GRV,  KC_1,    KC_2,    KC_3,  KC_4,  KC_5,  KC_6,  KC_7,   KC_8,  KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_DEL,
+    KC_PGUP, KC_TAB,  KC_Q,    KC_W,    KC_E,  KC_R,  KC_T,  KC_Y,  KC_U,   KC_I,  KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSPC,
+    KC_PGDN, KC_CAPS, KC_A,    KC_S,    KC_D,  KC_F,  KC_G,  KC_H,  KC_J,   KC_K,  KC_L,    KC_SCLN, KC_QUOT, KC_NUHS, KC_ENT,
+    KC_END,  KC_LSFT, KC_NUBS, KC_Z,    KC_X,  KC_C,  KC_V,  KC_B,  KC_N,   KC_M,  KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
+    KC_LGUI, KC_LCTL, KC_LGUI, KC_LALT,                             KC_SPC,                          KC_RALT, KC_APP,  KC_RCTL),
+
+    [1] = LAYOUT( /* FN1 */
```
Comment (noroadsleft) [inline keyboards/allison/keymaps/default/keymap.c]:
```suggestion
    [1] = LAYOUT_all( /* FN1 */
```

File: keyboards/allison/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+             RESET,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_UP,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_LEFT, KC_DOWN, KC_RGHT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, BL_TOGG, BL_INC,  BL_DEC,  BL_BRTG, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                                     KC_TRNS,                            KC_TRNS, KC_TRNS, KC_TRNS),
+
+    [2] = LAYOUT( /* Empty for dynamic keymaps */
```
Comment (noroadsleft) [inline keyboards/allison/keymaps/default/keymap.c]:
```suggestion
    [2] = LAYOUT_all( /* Empty for dynamic keymaps */
```

File: keyboards/allison/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+             KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                                     KC_TRNS,                            KC_TRNS, KC_TRNS, KC_TRNS),
+
+    [3] = LAYOUT( /* Empty for dynamic keymaps */
```
Comment (noroadsleft) [inline keyboards/allison/keymaps/default/keymap.c]:
```suggestion
    [3] = LAYOUT_all( /* Empty for dynamic keymaps */
```

File: keyboards/allison/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT( /* Base */
```
Comment (noroadsleft) [inline keyboards/allison/keymaps/via/keymap.c]:
```suggestion
    [0] = LAYOUT_all( /* Base */
```

File: keyboards/allison/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+             KC_ESC,  KC_F1,   KC_F2,   KC_F3, KC_F4, KC_F5, KC_F6, KC_F7,  KC_F8, KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_INS,
+    KC_HOME, KC_GRV,  KC_1,    KC_2,    KC_3,  KC_4,  KC_5,  KC_6,  KC_7,   KC_8,  KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSLS, KC_DEL,
+    KC_PGUP, KC_TAB,  KC_Q,    KC_W,    KC_E,  KC_R,  KC_T,  KC_Y,  KC_U,   KC_I,  KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSPC,
+    KC_PGDN, KC_CAPS, KC_A,    KC_S,    KC_D,  KC_F,  KC_G,  KC_H,  KC_J,   KC_K,  KC_L,    KC_SCLN, KC_QUOT, KC_NUHS, KC_ENT,
+    KC_END,  KC_LSFT, KC_NUBS, KC_Z,    KC_X,  KC_C,  KC_V,  KC_B,  KC_N,   KC_M,  KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
+    KC_LGUI, KC_LCTL, KC_LGUI, KC_LALT,                             KC_SPC,                          KC_RALT, KC_APP,  KC_RCTL),
+
+    [1] = LAYOUT( /* FN1 */
```
Comment (noroadsleft) [inline keyboards/allison/keymaps/via/keymap.c]:
```suggestion
    [1] = LAYOUT_all( /* FN1 */
```

File: keyboards/allison/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+             RESET,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_UP,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_LEFT, KC_DOWN, KC_RGHT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, BL_TOGG, BL_INC,  BL_DEC,  BL_BRTG, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                                     KC_TRNS,                            KC_TRNS, KC_TRNS, KC_TRNS),
+
+    [2] = LAYOUT( /* Empty for dynamic keymaps */
```
Comment (noroadsleft) [inline keyboards/allison/keymaps/via/keymap.c]:
```suggestion
    [2] = LAYOUT_all( /* Empty for dynamic keymaps */
```

File: keyboards/allison/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+             KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                                     KC_TRNS,                            KC_TRNS, KC_TRNS, KC_TRNS),
+
+    [3] = LAYOUT( /* Empty for dynamic keymaps */
```
Comment (noroadsleft) [inline keyboards/allison/keymaps/via/keymap.c]:
```suggestion
    [3] = LAYOUT_all( /* Empty for dynamic keymaps */
```

File: keyboards/allison/rules.mk [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+# Otherwise, delete this section
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
```
Comment (noroadsleft) [inline keyboards/allison/rules.mk]: Lines 14-23 should be deleted.

File: keyboards/allison/rules.mk [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (noroadsleft) [inline keyboards/allison/rules.mk]: Please align the comments in this block.

```Makefile
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```

File: keyboards/allison/allison.c [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "allison.h"
+
+// Optional override functions below.
```
[the remark this answers is not in the document]
Comment (yiancar) [reply keyboards/allison/allison.c]: Should we also fix the new_keyboard.sh so this doesnt get produced?

File: keyboards/allison/rules.mk (hunk 11 above)
Comment (yiancar) [reply keyboards/allison/rules.mk]: Should we also fix the new_keyboard.sh so this doesnt get produced?

Comment (fauxpark) [reply keyboards/allison/rules.mk]: I think so, yes. Generally it's going to be pretty rare that a board has a bootloader that isn't one of the ones we have listed.

File: keyboards/allison/allison.c (hunk 13 above)
[the remark this answers is not in the document]
Comment (noroadsleft) [reply keyboards/allison/allison.c]: We're kicking that idea around internally, actually. No decision has yet been made though.

File: keyboards/allison/keymaps/default/config.h [hunk 14]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2020 Yiancar
```
Comment (fauxpark) [inline keyboards/allison/keymaps/default/config.h]: This file can be deleted.

**Comment by yiancar** [comment]
done!

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**